internal/pkg/build/sources/oci_unpack_linux.go

```diff
@@ -68,5 +70,41 @@ func unpackRootfs(b *sytypes.Bundle, tmpfsRef types.ImageReference, sysCtx *type
 	os.RemoveAll(b.Rootfs())
 
 	// Unpack root filesystem
-	return umocilayer.UnpackRootfs(context.Background(), engineExt, b.Rootfs(), manifest, &mapOptions)
+	err = umocilayer.UnpackRootfs(context.Background(), engineExt, b.Rootfs(), manifest, &mapOptions)
+	if err != nil {
+		return err
+	}
+
+	// Fix permissions to allow the rootfs to be moved and deleted by the user
+	// if we performed a rootless extraction (#4517)
+	if mapOptions.Rootless {
+		sylog.Debugf("Fixing rootless permissions on %s", b.Rootfs())
+		return fixPermsRootless(b.Rootfs())
+	}
+
+	return nil
+
+}
+
+// fixPermsRootless forces permissions on the rootfs so that it can be easily
+// moved and deleted by a non-root user owner.
+func fixPermsRootless(rootfs string) (err error) {
+	err = filepath.Walk(rootfs, func(path string, f os.FileInfo, err error) error {
+		// Directories must have the owner 'rx' bits to allow traversal and reading on move, and the 'w' bit
+		// so their content can be deleted by the user when the rootfs/sandbox is deleted
+		switch mode := f.Mode(); {
+		case mode.IsDir():
+			if err := os.Chmod(path, f.Mode().Perm()|0700); err != nil {
+				return fmt.Errorf("error setting rootless sandbox permission for %s: %s", path, err)
```

Comment:
returning an error will will cause the walk to stop, so other files and directories won't be fixed, possibly leaving the directory in the undesired state. If you log the error but don't return it, Walk will continue fixing the rest of the directories.

Reply:
If we can't walk to or chmod any single file then it's very likely we have a fundamental issue leading to a totally broken rootfs... or at least one that will be un-moveable / copyable / removable for the user. From past experience with the singularity `2.x` C code and `3.x` `image-tools` patch I wrote to manage the extraction perms, breakage is all or nothing, and stopping early seems better than spitting out huge error logs that bust out of terminal scrollback on the 1000s of files that are in big containers.

I'd personally rather abandon all hope early, and maybe end up leave something lying in `/tmp` where it has a good chance of auto cleanup on the HPC systems that are generally using rootless sandboxes.... than let the code proceed to anything else.

Reply:
@ikaneshiro - can we page you for a third opinion?

Reply:
okay - I've come around to your thinking no @mem - am counting the errors but letting it try the fixes in current code (will push up in a bit).